### PR TITLE
fix account-recovery loading

### PR DESCRIPTION
**Pull request by VivekVinushanth**
#### Reason for this fix:
Since the Account Recovery is loaded as a component into security page has to employ placeholder loader.

#### After Fix 
- For at least one account recovery options

https://user-images.githubusercontent.com/32198547/129915964-a11c7029-d1c9-4b3a-be39-343c4b9eada7.mp4

- For no options available 

https://user-images.githubusercontent.com/32198547/129915475-75859c16-e181-4bf4-973a-bc811aa30c1a.mp4

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx [hunk 1]
```diff
@@ -150,7 +155,7 @@ export const AccountRecoveryComponent: FunctionComponent<AccountRecoveryProps> =
         getPreferences();
     }, []);
 
-    return (
+    return (!isAccountRecoveryDetailsLoading? (
```
Comment (yasinmiran) [inline apps/myaccount/src/components/account-recovery/account-recovery.tsx:158]:
```suggestion
    return (!isAccountRecoveryDetailsLoading ? (
```

**Review by yasinmiran — APPROVED** [review]

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx [hunk 2]
```diff
@@ -176,41 +187,50 @@ export const AccountRecoveryComponent: FunctionComponent<AccountRecoveryProps> =
...
+                                    </Placeholder.Header>
+                                    <Placeholder.Header image>
+                                        <Placeholder.Line />
+                                        <Placeholder.Line />
+                                    </Placeholder.Header>
+                                </Placeholder>
+                            </EmphasizedSegment>
                     )
-                    : null
+                    }
```
Comment (brionmario) [inline apps/myaccount/src/components/account-recovery/account-recovery.tsx:232]: Possible formatting error here. Lets reformat the ternary this way https://github.com/wso2/identity-apps/blob/master/docs/DEVELOPER.md#ternary-expressions

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx [hunk 3]
```diff
@@ -69,12 +70,13 @@ export const AccountRecoveryComponent: FunctionComponent<AccountRecoveryProps> =
...
     const [ isUsernameRecoveryEnabled, setIsUsernameRecoveryEnabled ] = useState<boolean>(false);
+    const [ isAccountRecoveryDetailsLoading, setIsAccountRecoveryDetailsLoading ] = useState<boolean>(false);
 
     /**
      * The following method gets the preference for account recovery.
      */
     const getPreferences = (): void => {
-
+        setIsAccountRecoveryDetailsLoading(true);
         const recoveryConnector: PreferenceRequest[] = [
```
Comment (brionmario) [inline apps/myaccount/src/components/account-recovery/account-recovery.tsx:80]:
```suggestion

        const recoveryConnector: PreferenceRequest[] = [
```

**Review by brionmario — APPROVED** [review]

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx (hunk 3 above)
Comment (VivekVinushanth) [reply apps/myaccount/src/components/account-recovery/account-recovery.tsx:81]: Addressed with fea49096450d560b42eb0cea808faec3c94da32d

File: apps/myaccount/src/components/account-recovery/account-recovery.tsx (hunk 2 above)
Comment (VivekVinushanth) [reply apps/myaccount/src/components/account-recovery/account-recovery.tsx:230]: Addressed with 1997335a9368d5f07d34a552c95a7c0045369740

**Pull request merged**